fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -946,11 +946,24 @@ private void handleShowVariables() {
     // Show create database
     private void handleShowCreateDb() throws AnalysisException {
         ShowCreateDbStmt showStmt = (ShowCreateDbStmt) stmt;
+        String catalogName = showStmt.getCatalogName();
+        String dbName = showStmt.getDb();
         List<List<String>> rows = Lists.newArrayList();
-        Database db = connectContext.getGlobalStateMgr().getDb(showStmt.getDb());
-        MetaUtils.checkDbNullAndReport(db, showStmt.getDb());
-        rows.add(Lists.newArrayList(showStmt.getDb(),
-                "CREATE DATABASE `" + showStmt.getDb() + "`"));
+
+        if (Strings.isNullOrEmpty(catalogName) || CatalogMgr.isInternalCatalog(catalogName)) {
```

Comment:
why not get catalog name from session

Reply:
I think they are equivalent.

Reply:
So when is catalog name null?

Reply:
`catalogName` is unset for internal database

Reply:
You need to fill database and catalog in the ShowStmtAnalyzer.visitShowCreateDbStatement()

Reply:
filled in BasicDbStmtAnalyzer.BasicDbStmtAnalyzerVisitor#visitShowCreateDbStatement